design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+| conjur.org/secret-file-path.{secret-group}     | File path (including file name) for secrets file to be written. Default is `{secret-group}.{secret-file-type}`, e.g. `database.yaml`. |
+| conjur.org/secret-file-type.{secret-group}     | Allowed values:<ul><li>yaml (default)</li><li>json</li><li>dotenv</li><li>bash</li></ul>This setting is ignored when `conjur.org/secret-file-template.{secret-group}` is configured. |
+| conjur.org/secret-file-perms.{secret-group}    | File permissions for secrets file to be written, in octal format. Defaults to `660` |
+| conjur.org/secret-file-template.{secret-group} | Customized secret file template using the Golang templating language. |
+
+###### Example `conjur-secrets` List Without Aliases
+
+The following is an example of a `conjur.org/conjur-secrets.{secret-group}`
+Annotation that defines a flat (i.e. no aliases) list of Conjur variable paths
+for a hypothetical secrets group named `database`:
+
+```
+    conjur.org/conjur-secrets.database: |
+      - prod/backend/url
+      - prod/backend/port 
+      - prod/backend/password
+      - prod/backend/username
+```
+
+For `conjur-secrets` entries without aliases, the last word in the path is
+used as the application secret name. So for the example above, if
+`conjur.org/secret-file-type.database` is set to `yaml`, then the resulting
+secrets file that is written to the shared secrets volume might look similar
+to the following:
+
+```
+url: https://database.example.com
+port: 5
+password: my-secret-p@$$w0rd
+username: postgres
+```
+
+###### Example `conjur-secrets` List With Aliases and `conjur-secrets-path` Prefix
+
+The following is an example of a `conjur.org/conjur-secrets.{secret-group}`
+Annotation that defines a list of Conjur variable paths, including some that
+use an alias, for a hypothetical secrets group named `cache`:
+
+```
+    conjur.org/conjur-secrets.cache: |
+      - url
+      - admin-password: password
+      - admin-username: username
+
+    conjur.org/conjur-secrets-path.cache: dev/memcached/
+
+```
+
+In this example, the paths for the secrets that are retrieved from Conjur are:
+
+- dev/memcached/url
+- dev/memcached/password
+- dev/memcached/username
+
+and the names of the secrets that are written to the secrets file are:
+
+- url
+- admin-password
+- admin-username
+
+###### Example `conjur.org/secret-file-template.{secret-group}`
+
+The following is an example of a custom secrets file template that uses the
+Golang templating language. Variables can be referenced via their name
+(their alias if provided, otherwise, the last word in their Conjur variable
+path) and are replaced with their value from Conjur:
+
+```
+    conjur.org/secret-template-custom.cache: |
+      {
+        "cache": {
+          "url": {{ .url }},
+          "password": {{ .admin-password }},
+          "username": {{ .admin-username }},
+          "port": 123456
+        }
+      }
+```
+
```

Comment:
maybe include the sample output file here?

also, if no output file name is given this will write to a file named `cache.yaml`. I'd recommend we _require_ output file name if custom templates are used. what do you think?

Reply:
Do we have an error case for when the template is given but no file path is specified? If not, we should add one.

Reply:
added a suggestion.